Mathlib/Order/Concept.lean

```diff
@@ -191,6 +200,101 @@ theorem lowerPolar_anti : Antitone (lowerPolar r) :=
 @[deprecated (since := "2025-07-10")]
 alias extentClosure_anti := lowerPolar_anti
 
+theorem lowerPolar_upperPolar_monotone : Monotone (lowerPolar r ∘ upperPolar r) :=
+  (lowerPolar_anti r).comp (upperPolar_anti r)
+
+theorem upperPolar_lowerPolar_monotone : Monotone (upperPolar r ∘ lowerPolar r) :=
+  (upperPolar_anti r).comp (lowerPolar_anti r)
+
+/-- The `extentClosure` of a set is the smallest extent containing it. -/
+@[simps!]
+def extentClosure (r : α → β → Prop) : ClosureOperator (Set α) :=
+  (gc_upperPolar_lowerPolar r).closureOperator
+
+/-- The `intentClosure` of a set is the smallest intent containing it. -/
+@[simps!]
+def intentClosure (r : α → β → Prop) : ClosureOperator (Set β) :=
+  (gc_lowerPolar_upperPolar r).closureOperator
+
+/-! ### `IsIntent` and `IsExtent` -/
+
+
+namespace Set
+
+variable {r}
+
+/-- A set is an extent when either of the following equivalent definitions holds:
+
+- The `lowerPolar` of its `upperPolar` is itself.
+- The set is the `lowerPolar` of some other set. -/
+def IsExtent (r : α → β → Prop) (s : Set α) := lowerPolar r (upperPolar r s) = s
```

Comment:
I would advise against putting this definition in the `Set` namespace since it is not property of bare sets, but a property of sets with an ambient relation. I think `Order.IsExtent` is a fine name